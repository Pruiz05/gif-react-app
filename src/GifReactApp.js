import React, { useState } from "react";
import { AddCategory } from "./components/AddCategory";

import { GifGrid } from "./components/GifGrid";

const GifReactApp = ({defaultCategories = []}) => {
  // const categories = ['One piece', 'Bleach', 'Naruto'];

  const [categories, setCategories] = useState(defaultCategories);
 
//   const handleAdd = () =>{
//       setCategories([...categories,'Hunter x Hunter']);
//   }

  return (
    <>
      <h2>Gif React App</h2>
      <AddCategory setCategories={setCategories} />
      <hr />
      {/* <button onClick={handleAdd}>Agregar</button> */}
      {/* <ol> */}
        {categories.map((category) => (
          <GifGrid key={category} category={category} />
        ))}
    
    </>
  );
};

export default GifReactApp;
