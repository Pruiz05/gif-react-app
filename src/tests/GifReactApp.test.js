import { shallow } from "enzyme";
import GifReactApp from "../GifReactApp";

describe("Pruebas en <GifReactApp />", () => {
  test("debe de mostrarse correctamente", ( ) => {
    const wrapper = shallow(<GifReactApp />);

    expect(wrapper).toMatchInlineSnapshot();
  });

  test('debe de mostrar una lista de categorias', () => {
      const categories = ['Bleach', 'One Piece']

      const wrapper = shallow(<GifReactApp defaultCategories={categories}/>)

      expect(wrapper).toMatchSnapshot()

      expect(wrapper.find('GifGrid').length).toBe(categories.length)
  })
  
});
 