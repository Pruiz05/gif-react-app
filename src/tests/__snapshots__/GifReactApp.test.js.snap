// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifReactApp /> debe de mostrar una lista de categorias 1`] = `
<Fragment>
  <h2>
    Gif React App
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <GifGrid
    category="Bleach"
    key="Bleach"
  />
  <GifGrid
    category="One Piece"
    key="One Piece"
  />
</Fragment>
`;
